R/derive_param_wbc_abs.R

```diff
@@ -0,0 +1,162 @@
+#' \deqn{\frac{White Blood Cell Count  * Percentage Value}{100}}
+#' New records are created for each group of records (grouped by `by_vars`) if 1) the white blood
+#' cell component in absolute value is not already available from the input dataset, and 2) the
+#' white blood cell absolute value (identified by `wbc_code`) and the white blood cell differential
+#' (identified by `diff_code`) are both present.
+#'
+#' @author Annie Yang
+#'
+#' @return The input dataset with the new parameter added
+#'
+#' @keywords derivation bds adlb
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#' test_lb <- tibble::tribble(
+#' ~USUBJID, ~PARAMCD, ~AVAL, ~PARAM, ~VISIT,
+#' "P01",    "WBC",     33,   "Leukocyte Count (10^9/L)",       "CYCLE 1 DAY 1",
+#' "P01",    "WBC",     38,   "Leukocyte Count (10^9/L)",       "CYCLE 2 DAY 1",
+#' "P01",    "LYMLE",   90,   "Lymphocytes (fraction of 1)",    "CYCLE 1 DAY 1",
+#' "P01",    "LYMLE",   70,   "Lymphocytes (fraction of 1)",    "CYCLE 2 DAY 1",
+#' "P01",    "ALB",     36,   "Albumin (g/dL)",                 "CYCLE 2 DAY 1",
+#' "P02",    "WBC",     33,   "Leukocyte Count (10^9/L)",       "CYCLE 1 DAY 1",
+#' "P02",    "LYMPH",   29,   "Leukocyte Count (10^9/L)",       "CYCLE 1 DAY 1",
+#' "P02",    "LYMLE",   87,   "Lymphocytes (fraction of 1)",    "CYCLE 1 DAY 1",
+#' "P03",    "LYMLE",   89,   "Lymphocytes (fraction of 1)",    "CYCLE 1 DAY 1"
+#' )
+#'
+#' derive_param_wbc_abs (dataset = test_lb,
+#'                       by_vars = vars(USUBJID, VISIT),
+#'                       set_values_to = vars(
+#'                         PARAMCD = "LYMPH",
+#'                         PARAM = "Lymphocytes Abs (10^9/L)",
+#'                         DTYPE = "CALCULATION"
+#'                       ),
+#'                       get_unit_expr = extract_unit(PARAM),
+#'                       wbc_code = "WBC",
+#'                       diff_code = "LYMLE")
+
+derive_param_wbc_abs <- function(dataset,
+                                 by_vars,
+                                 set_values_to,
+                                 get_unit_expr,
+                                 wbc_unit = "10^9/L",
+                                 diff_unit = "fraction of 1",
+                                 wbc_code = "WBC",
+                                 diff_code) {
+  assert_vars(by_vars)
+  assert_data_frame(dataset, required_vars = vars(!!!by_vars, PARAMCD, AVAL))
+  assert_character_scalar(wbc_code)
+  assert_character_scalar(diff_code)
+  get_unit_expr <- assert_expr(enquo(get_unit_expr))
+  assert_character_scalar(wbc_unit)
+  assert_character_scalar(diff_unit)
+  by_vars_str <- vars2chr(by_vars)
+
+  # Check unit of input parameters
+  assert_unit(dataset, wbc_code, required_unit = wbc_unit, get_unit_expr = !!get_unit_expr)
+  assert_unit(dataset, diff_code, required_unit = diff_unit, get_unit_expr = !!get_unit_expr)
+
+  # select observations and variables required for new observations.
```

Comment:
Comments should start with Proper Case.  I think styler fixes things like this if you run it on the selection of code.

Reply:
Thanks, I capitalized the first word of the comments, or did you mean to capitalize each word?

Reply:
Just the first word - I think it to sytler if fixes that kind of thing - but not sure